### ISSUE TITLE
UNFs should be displaying in citation for Murray Dataverse (and more) datasets

### ISSUE THREAD

**Issue by sbarbosadataverse** [IQSS/dataverse#2327]
Datasets with tabular data not displaying UNF in citation

**Comment by eaquigley** [comment]
@landreev During this week's github review meeting, we realized this issue was never assigned to you for 4.4 when it should have been. Can you please look at this and fix it.

**Comment by djbrooke** [comment]
I'm going to move this from 4.4 and into 4.5.

**Comment by sbarbosadataverse** [comment]
FYI: this is not only Murray data, but Gary's data are impacted as well.
And I'm certain he has not noticed the UNF is missing but won't be happy
when he does find this out. Why are we moving it to 4.5?
Thanks

On Wed, Jun 22, 2016 at 12:06 PM, Danny Brooke <notifications@github.com>
wrote:

> I'm going to move this from 4.4 and into 4.5.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_IQSS_dataverse_issues_2327-23issuecomment-2D227793295&d=CwMCaQ&c=WO-RGvefibhHBZq3fL85hQ&r=8R6PzVqt1PEocauQgZMGXsGz29-nb19M7eqlo1d8EVs&m=g7sOqiFsK_MTlUMbfVjzxL5wMVt3Xdtcc0xNQ78_D0w&s=SELZdZvURgLil0YEwo682_hurmVUdqRq4Cb7xONDlh4&e=>,
> or mute the thread
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe_AH79KgRDuqnGiBIK1t659e1zVeFCDcWMks5qOV2ggaJpZM4FUDik&d=CwMCaQ&c=WO-RGvefibhHBZq3fL85hQ&r=8R6PzVqt1PEocauQgZMGXsGz29-nb19M7eqlo1d8EVs&m=g7sOqiFsK_MTlUMbfVjzxL5wMVt3Xdtcc0xNQ78_D0w&s=CyXLod4gyvWRRZ0U8SESZ1VokH41wnIr_9VQfFHg-Uw&e=>
> .
>



-- 
Sonia Barbosa
Manager of Data Curation, IQSS Dataverse Network
Manager of the Murray Research Archive, IQSS
Data Science
Harvard University

Dataverse 4.0 is now available for use!
http://dataverse.harvard.edu

All test dataverses should be created  in 4.0 Demo!
http://dataverse-demo.iq.harvard.edu/

Join our Dataverse Community!
https://groups.google.com/forum/#!forum/dataverse-community

**Comment by djbrooke** [comment]
I'd have liked to get this in as well, but we are very close to delivering the widgets part of this release and I'm hesitant to try and get in changes that haven't yet been started. It's unfortunate that this was not brought in and worked on (I should have absolutely noted this earlier), but I'd like to move forward without it in 4.4 and I'll make sure to do better track things in the future.

**Comment by sbarbosadataverse** [comment]
Absolutely your call, but wanted you to know why it was so important to
begin with. Gary really makes use of this UNFs so I suggest we keep this in
mind for 4.5.
 thnks

On Wed, Jun 22, 2016 at 2:02 PM, Danny Brooke <notifications@github.com>
wrote:

> I'd have liked to get this in as well, but we are very close to delivering
> the widgets part of this release and I'm hesitant to try and get in changes
> that haven't yet been started. It's unfortunate that this was not brought
> in and worked on (I should have absolutely noted this earlier), but I'd
> like to move forward without it in 4.4 and I'll make sure to do better
> track things in the future.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_IQSS_dataverse_issues_2327-23issuecomment-2D227827252&d=CwMCaQ&c=WO-RGvefibhHBZq3fL85hQ&r=8R6PzVqt1PEocauQgZMGXsGz29-nb19M7eqlo1d8EVs&m=1C1-tmZXy-frCkWbCKQlHIa2AfNN2pb3fJxB38anhSk&s=26KdWrTlfuZsdHVyZ5GgE4a1TtGI56gUORn-mVjmbvM&e=>,
> or mute the thread
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe_AH79KoIDReFn6e21jsIvuBUMzSc6hrmjks5qOXi9gaJpZM4FUDik&d=CwMCaQ&c=WO-RGvefibhHBZq3fL85hQ&r=8R6PzVqt1PEocauQgZMGXsGz29-nb19M7eqlo1d8EVs&m=1C1-tmZXy-frCkWbCKQlHIa2AfNN2pb3fJxB38anhSk&s=oyoGrN9aawVCkDGq25fENxSL0Lv0k1bZ2yA8EYhPR8Q&e=>
> .
>



-- 
Sonia Barbosa
Manager of Data Curation, IQSS Dataverse Network
Manager of the Murray Research Archive, IQSS
Data Science
Harvard University

Dataverse 4.0 is now available for use!
http://dataverse.harvard.edu

All test dataverses should be created  in 4.0 Demo!
http://dataverse-demo.iq.harvard.edu/

Join our Dataverse Community!
https://groups.google.com/forum/#!forum/dataverse-community

**Comment by djbrooke** [comment]
Checked with @landreev - he is going to work on this for 4.5.

**Comment by djbrooke** [comment]
Thanks @pdurbin for offering to take a look!

**Comment by pdurbin** [comment]
@djbrooke asked me to take a look at this issue. From looking at https://dataverse.lib.virginia.edu/dataverse/uva?q=protein and https://dataverse.lib.virginia.edu/dataset.xhtml?persistentId=doi:10.18130/V3/LWPBIY (screenshots below) I can see how UNFs looked in 4.3.1. For example:

`Shupnik, Margaret, 2016, "A new role for Wilms Tumor protein 1: differential activities of + KTS and -KTS variants to regulate LHβ transcription", http://dx.doi.org/10.18130/V3/LWPBIY, University of Virginia Dataverse, V1 [UNF:6:2NpcWLcfWHxiPtaTlY+rkQ==]`

![screen shot 2016-07-26 at 10 35 13 am](https://cloud.githubusercontent.com/assets/21006/17142088/cb5085d4-531c-11e6-971c-a8bac48b9778.png)
![screen shot 2016-07-26 at 10 35 17 am](https://cloud.githubusercontent.com/assets/21006/17142087/cb4ebc68-531c-11e6-9e17-c5eac9a0e9f3.png)

**Comment by djbrooke** [comment]
At standup today, @landreev mentioned that he is already working on this. Reassigning. Thanks @pdurbin for being willing to work on this!

**Comment by sekmiller** [comment]
As discussed I took care of the issue where datasets with UNFs were not having them carried forward when a new dataset version is created (by any means other than adding a new file.)

**Comment by landreev** [comment]
OK, just to summarize: 
This was not a matter of the UNF *not being shown* in the displayed citation; 
It was a matter of not copying the UNF of an existing version to the draft, if the said draft was created by means other than adding another tabular file. 
The UNF would be properly calculated eventually, when and if another tab. file were added to the dataset. 

The issue has now been solved, not by me, but by Steven. The fix is in the branch 3221-unf-fixes-4.5; same branch where I pushed my code for the "recalculate-the-unf-when-tab-files-are-deleted" fix. 

It is ready for QA; I just want to double check what the procedure is, for 4.5.
